python-lib/tagger.py

```diff
@@ -1,9 +1,110 @@
+"""Module to prepare the tagging of the documents
+splits on sentences and instanciates a Matcher to tag the documents"""
+
+from spacy_tokenizer import MultilingualTokenizer
+from dkulib_io_utils import generate_unique
+from spacy.matcher import PhraseMatcher
+from fastcore.utils import store_attr
+from typing import AnyStr
 import pandas as pd
 
 
 class Tagger:
-    def __init__(self, settings):
-        pass
+    def __init__(
+        self,
+        text_df,
+        ontology_df,
+        text_column,
+        language,
+        language_column,
+        tag_column,
+        category_column,
+        keyword_column,
+        lemmatize,
+        case_sensitive,
+        normalize,
+        mode,
+    ):
+        store_attr(but=["text_df", "ontology_df"])
+        self.text_df = text_df.get_dataframe()
+        self.ontology_df = ontology_df.get_dataframe()
+        self.matcher_dict = {}
+        self.nlp_dict = {}
+        self._matching_pipeline()
+
+    def _get_patterns(self):
+        """
+        Public function called in tagger.py
+        Creates the list of patterns :
+        - If there aren't category -> list of the keywords (string list)
+        - If there are categories  -> list of dictionaries, {"label": category, "pattern": keyword}
+        """
+        list_of_tags = self.ontology_df[self.tag_column].values.tolist()
+        list_of_keywords = self.ontology_df[self.keyword_column].values.tolist()
+        self.keyword_to_tag = dict(zip(list_of_keywords, list_of_tags))
+
+        if self.category_column:
+            list_of_categories = self.ontology_df[self.category_column].values.tolist()
+            self.patterns = [
+                {"label": label, "pattern": pattern}
+                for label, pattern in zip(list_of_categories, list_of_keywords)
+            ]
+        else:
+            self.patterns = list_of_keywords
+
+    def _list_sentences(self, row: pd.Series):
+        """Auxiliary function called in _matching_pipeline
+        Applies sentencizer and return list of sentences"""
+        return [
+            sentence.text
+            for sentence in self.nlp_dict[self.language](row[self.text_column]).sents
+        ]
+
+    def _matching_method_no_category(self, language: AnyStr):
```

Comment:
Overall advice for methods and function names: 
- use action verbs at the infinitive
- avoid adding words like `method`, `procedure` or `function`, as this information is already coded by the fact it's a `def` defined in a class